backend/src/test/java/com/ody/notification/service/NotificationServiceTest.java

```diff
@@ -0,0 +1,64 @@
+package com.ody.notification.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+
+import com.ody.common.BaseServiceTest;
+import com.ody.common.Fixture;
+import com.ody.mate.domain.Mate;
+import com.ody.mate.domain.Nickname;
+import com.ody.mate.repository.MateRepository;
+import com.ody.meeting.domain.Meeting;
+import com.ody.meeting.repository.MeetingRepository;
+import com.ody.member.domain.Member;
+import com.ody.member.repository.MemberRepository;
+import com.ody.route.domain.DepartureTime;
+import com.ody.route.domain.RouteTime;
+import com.ody.route.service.RouteService;
+import java.time.LocalDateTime;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.BDDMockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+
+class NotificationServiceTest extends BaseServiceTest {
+
+    @SpyBean
+    private NotificationService notificationService;
+
+    @MockBean
+    private RouteService routeService;
+
+    @Autowired
+    private MemberRepository memberRepository;
+
+    @Autowired
+    private MeetingRepository meetingRepository;
+
+    @Autowired
+    private MateRepository mateRepository;
+
+    @DisplayName("알림 생성 시점이 전송 시점보다 늦은 경우 즉시 전송된다")
+    @Test
+    void sendImmediatelyIfDepartureTimeIsPast() {
+        Member member = memberRepository.save(Fixture.MEMBER1);
+        Meeting meeting = meetingRepository.save(Fixture.ODY_MEETING1);
+        Mate mate = mateRepository.save(new Mate(meeting, member, new Nickname("제리"), Fixture.ORIGIN_LOCATION));
+
+        LocalDateTime today = LocalDateTime.now();
+        DepartureTime pastDepartureTime = new DepartureTime(new RouteTime(1), today.minusDays(1));
+
+        BDDMockito.given(routeService.calculateDepartureTime(any(), any(), any()))
+                .willReturn(pastDepartureTime);
+
+        notificationService.saveAndSendDepartureReminder(meeting, mate, member.getDeviceToken());
```

Comment:
assertThat 구문에 when절에 대한 코드를 넣기 보다는 바깥으로 추출되서 사용되는게 개인적으로는 더 깔끔하지 않나 생각이 드는데 조조는 어떠신가요 ?
```suggestion
               Optional<Notification> departureNotification = allMeetingLogs.stream()
                .filter(notification -> notification.getSendAt().getDayOfMonth() == today.getDayOfMonth())
                .findFirst();

        assertThat(departureNotification).isPresent();
```

Reply:
[반영 완료]
동의합니다~🙋‍♀️